include/one_dimensional_linear_interpolator.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+#define __ONE_DIMENSIONAL_LINEAR_INTERPOLATOR_H__
+
+#include "interpolator.hpp"
+#include <CL/sycl.hpp>
+#include <mpi.h>
+#include <neso_particles.hpp>
+#include <vector>
+
+using namespace NESO::Particles;
+using namespace cl;
+namespace NESO {
+
+/**
+ *  Class used to output a vector of y values, given some x values, based
+ * on provided x,y values for input
+ */
+
+class OneDimensionalLinearInterpolator : public Interpolator {
+public:
+  OneDimensionalLinearInterpolator(std::vector<double> x_data,
+                                   std::vector<double> y_data,
+                                   SYCLTargetSharedPtr sycl_target)
+      : Interpolator(x_data, y_data, sycl_target){
+
+        };
+
+  virtual void get_y(std::vector<double> &x_input,
+                     std::vector<double> &y_output) {
+    interpolate(x_input, y_output);
+  }
+
+protected:
+  virtual void interpolate(std::vector<double> &x_input,
+                           std::vector<double> &y_output) {
+
+    // testing error handling of interpolator
+    ErrorPropagate ep(m_sycl_target);
+    // sycl code
+    auto k_ep = ep.device_ptr();
+    ep.reset();
+    y_output = std::vector<double>(x_input.size());
+    sycl::buffer<double, 1> buffer_x_data(m_x_data.data(),
+                                          sycl::range<1>{m_x_data.size()});
+    sycl::buffer<double, 1> buffer_y_data(m_y_data.data(),
+                                          sycl::range<1>{m_y_data.size()});
+    sycl::buffer<double, 1> buffer_x_input(x_input.data(),
+                                           sycl::range<1>{x_input.size()});
+    sycl::buffer<double, 1> buffer_y_output(y_output.data(),
+                                            sycl::range<1>{y_output.size()});
+    sycl::buffer<double, 1> buffer_dydx(dydx.data(),
+                                        sycl::range<1>{dydx.size()});
+    const int m_x_data_size = m_x_data.size();
+    const int x_input_size = x_input.size();
+    auto event_interpolate =
+        this->m_sycl_target->queue.submit([&](sycl::handler &cgh) {
+          auto x_data_sycl =
+              buffer_x_data.get_access<sycl::access::mode::read>(cgh);
+          auto y_data_sycl =
+              buffer_y_data.get_access<sycl::access::mode::read>(cgh);
+          auto x_input_sycl =
+              buffer_x_input.get_access<sycl::access::mode::read>(cgh);
+          auto y_output_sycl =
+              buffer_y_output.get_access<sycl::access::mode::read_write>(cgh);
+          auto dydx_sycl =
+              buffer_dydx.get_access<sycl::access::mode::read>(cgh);
+          cgh.parallel_for<>(
+              sycl::range<1>(x_input_size), [=](sycl::id<1> idx) {
+                int k;
+                for (int i = 0; i < m_x_data_size; i++) {
+                  // error handling
+                  if (x_input_sycl[int(idx)] < x_data_sycl[0] or
```

Comment:
Slightly stylistic - I always put constant values like `x_input_sycl[int(idx)]` in a temporary like 
```
const double x = x_input_sycl[int(idx)];
```
to ensure a reload doesn't happen due to possible aliasing (or compiler fear of aliasing).